src/transactions/feed/queryHelper.ts

```diff
@@ -107,76 +125,65 @@ export function useFetchTransactions(): QueryHookResult {
-    Logger.info(TAG, `Fetched ${isPollResult ? 'new' : 'next page of'} transactions`)
-
-    for (const [networkId, result] of Object.entries(results) as Array<
-      [NetworkId, QueryResponse]
-    >) {
-      const returnedTransactions = result.data?.tokenTransactionsV3?.transactions ?? []
-
-      const returnedPageInfo = result.data?.tokenTransactionsV3?.pageInfo ?? null
-
-      // the initial feed fetch is from polling, exclude polled updates from that scenario
-      const isPolledUpdate = isPollResult && fetchedResult.pageInfo[networkId] !== null
-
-      if (returnedTransactions.length || returnedPageInfo?.hasNextPage) {
-        setFetchedResult((prev) => ({
-          transactions: deduplicateTransactions(prev.transactions, returnedTransactions),
-          // avoid updating pageInfo and hasReturnedTransactions for polled
-          // updates, as these variables are used for fetching the next pages
-          pageInfo: isPolledUpdate
-            ? prev.pageInfo
-            : { ...prev.pageInfo, [networkId]: returnedPageInfo },
-          hasTransactionsOnCurrentPage: isPolledUpdate
-            ? prev.hasTransactionsOnCurrentPage
-            : {
-                ...prev.hasTransactionsOnCurrentPage,
-                [networkId]: returnedTransactions.length > 0,
-              },
-        }))
-
-        if (isPollResult && returnedTransactions.length) {
-          // We store the first page in redux to show them to the users when they open the app.
-          // Filter out now empty transactions to avoid redux issues
-          const nonEmptyTransactions = returnedTransactions.filter(
-            (returnedTransaction) => !isEmpty(returnedTransaction)
-          )
-          const knownTransactionHashes = transactionHashesByNetwork[networkId]
-          let hasNewTransaction = false
-
-          // Compare the new tx hashes with the ones we already have in redux
-          for (const tx of nonEmptyTransactions) {
-            if (!knownTransactionHashes || !knownTransactionHashes.has(tx.transactionHash)) {
-              hasNewTransaction = true
-              break // We only need one new tx justify a refresh
-            }
-          }
-          // If there are new transactions update transactions in redux and fetch balances
-          if (hasNewTransaction) {
-            dispatch(updateTransactions(networkId, nonEmptyTransactions))
-            vibrateSuccess()
-          }
-        }
-      }
-    }
-  }
-
   const handleError = (error: Error) => {
     Logger.error(TAG, 'Error while fetching transactions', error)
   }
 
   // Query for new transaction every POLL_INTERVAL
   const { loading, error } = useAsync(
     async () => {
-      const result = await queryTransactionsFeed({
+      await queryTransactionsFeed({
         address,
         localCurrencyCode,
         params: allowedNetworkIds.map((networkId) => {
           return { networkId }
         }),
+        updateTransactionState: (networkId, result) => {
```

Comment:
I wonder if we need to worry that is is handled only for the "current" promise of `queryTransactionsFeed`.

But maybe since the result is deduplicated it's fine?

Reply:
We also handle the pending states for polling and pagination separately so it should only deduplicate on each one of the pending states respectively.